### DISCUSSION TITLE
what do I need to do with read_db ?

### DISCUSSION THREAD

**Discussion by ThomasGeroudet** [The-OpenROAD-Project/OpenROAD#6020] [Q&A]
Hello,

I have tried to load an existing db using :
read_db file.odb

But then I realized that some commands complain about liberty data not available.
So I need to load .lib file.
Do I need to do it before or after read_db ?
Do I need to link the design then ?
Are there other files that need to be read before or immediately after read_db ? lef, rc files, sdc, ... ?

Have a nice day,
Thomas

**Comment by maliberty** [comment]
It is easiest to use a flow like [OpenROAD-flow-scripts](https://openroad-flow-scripts.readthedocs.io/en/latest/index2.html) that automates all this knowledge.

**Reply by QuantamHD** [reply]
You need to read_db, then read your lib, set_rcs and then SDC in order to get timing information.

**Reply by maliberty** [reply]
Also estimate parasitics, or read a spef.  In order to estimate you need to setup the rc values. Etc.

You should read the liberty before the odb.

**Reply by ThomasGeroudet** [reply]
what about tech lef and library lef ? are they included in the odb ?

**Reply by maliberty** [reply]
Yes

**Comment by titan73** [comment]
Shouldn't the dependencies be saved to the database so they can be reloaded automatically when reloaded as some tools do?

**Reply by maliberty** [reply]
That would be nice to add.  Note that you might not always want to reload them, eg I'm looking at layout and don't care about timing right now and want to save startup time.
